test/search_index_ddoc_update_tests.erl

```diff
@@ -0,0 +1,277 @@
+    fake_clouseau(),
+    fake_dreyfus_index(),
+
+    Host = host_url(PortType),
+    upload_ddoc(Host, ?b2l(DbName), ?DDOC),
+    {Host, ?b2l(DbName)}.
+
+teardown(Ctx) ->
+    ok = config:delete("admins", ?USER, _Persist=false),
+    (catch meck:unload(clouseau_rpc)),
+    (catch meck:unload(dreyfus_index)),
+    test_util:stop_couch(Ctx).
+
+teardown(PortType, {_Host, DbName}) ->
+    (catch meck:unload(mochiweb_socket)),
+    (catch meck:unload(clouseau_rpc)),
+    (catch meck:unload(dreyfus_index)),
+    delete_db(PortType, ?l2b(DbName)),
+    ok.
+
+search_test_() ->
+    {
+        "Check search index closes after delete",
+        {
+            setup,
+            fun start/0, fun teardown/1,
+            [
+                make_test_case(clustered, [fun should_stop_indexing_after_deleting_index/2]),
+                make_test_case(clustered, [fun should_stop_indexing_after_deleting_ddoc/2])
+            ]
+        }
+    }.
+
+fake_clouseau() ->
+    ok = meck:new([clouseau_rpc], [non_strict]),
+    ok = meck:expect(clouseau_rpc, open_index, ['_', '_', '_'], {ok, self()}),
+    ok = meck:expect(clouseau_rpc, await, ['_', '_'], ok),
+    ok = meck:expect(clouseau_rpc, get_update_seq, ['_'], {ok, 10}),
+    ok = meck:expect(clouseau_rpc, info, ['_'], {ok, [{disk_size, 10},
+      {doc_count, 10},
+      {doc_del_count, 0},
+      {pending_seq, 11},
+      {committed_seq, 10}]}),
+    ok = meck:expect(clouseau_rpc, search, ['_', '_'], {ok, [
+              {update_seq, 10},
+              {total_hits, 0},
+              {hits, []}
+            ]}),
+    ok = meck:expect(clouseau_rpc, update, ['_', '_', '_'], ok),
+    ok = meck:expect(clouseau_rpc, commit, ['_', '_'], ok),
+    ok = meck:expect(clouseau_rpc, search, ['_', '_', '_', '_', '_', '_'], {ok, [
+              {update_seq, 10},
+              {total_hits, 0},
+              {hits, []}
+            ]}).
+
+fake_dreyfus_index() ->
+    ok = meck:new([dreyfus_index], [passthrough]),
+    ok = meck:expect(dreyfus_index, await, ['_', '_'], ok),
+    ok = meck:expect(dreyfus_index, search, ['_', '_'], {ok, #top_docs{
+               update_seq = 10,
+               total_hits = 0,
+               hits = []
+              }}).
+
+make_test_case(Mod, Funs) ->
+    {
+        lists:flatten(io_lib:format("~s", [Mod])),
+        {foreachx, fun setup/1, fun teardown/2, [{Mod, Fun} || Fun <- Funs]}
+    }.
+
+should_stop_indexing_after_deleting_index(PortType, {Host, DbName}) ->
+    {timeout, 60,
+     ?_test(begin
+         ok = meck:reset(clouseau_rpc),
+         ok = meck:reset(dreyfus_index),
+         ok = create_doc(PortType, ?l2b(DbName), <<"doc_id">>, {[]}),
+         ReqUrl = Host ++ "/" ++ DbName ++ "/_design/searchddoc/_search/other?q=*:*",
+         {ok, Status, _Headers, _Body} =
+              test_request:get(ReqUrl, [?AUTH]),
+          ?assertEqual(200, Status),
+          DDocId = "_design/searchddoc",
+          Sig = ensureIndexIsOpened(DbName, DDocId, "other"),
```

Comment:
Yes, the code in [here](https://github.com/cloudant-labs/dreyfus/blob/0739e2329877ea9222c8b00340c9d3b37b423220/src/dreyfus_index_manager.erl#L123-L132) is taking care of handling the event in all db shards.